src/error.rs

```diff
@@ -1195,6 +1205,13 @@ impl ToDiagnostic<FileId> for ParseError {
                         .join(",")
                 ))
                 .with_labels(vec![primary(span)]),
+            ParseError::InvalidUniRecord(illegal_span, tail_span, span) => Diagnostic::error()
+                .with_message(format!("invalid record literal"))
+                .with_labels(vec![
+                    primary(span),
+                    secondary(illegal_span).with_message("can't use non-type construct"),
```

Comment:
Similarly, I doubt that “can't use non-type construct” would make for a useful error message.